### ISSUE TITLE
This doesn't work

### ISSUE THREAD

**Issue by fschoenfeldt** [cichun/expo-detox-test#1]
I've attached a video demonstrating that Detox doesn't open the actual React App in Expo:

https://user-images.githubusercontent.com/4721311/126435671-17215892-507e-4482-a76e-214d4043cca8.mov

I noticed that when installing the lockfile changes:
```diff
diff --git a/yarn.lock b/yarn.lock
index 2d86ce6..7b772b8 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -4240,10 +4240,10 @@ detox-expo-helpers@^0.6.0:
     semver "^5.6.0"
     xdl "^51.5.0"

-detox@~17.14.3:
-  version "17.14.3"
-  resolved "https://registry.yarnpkg.com/detox/-/detox-17.14.3.tgz#df83152eac01966a0682c6def9c2da22a3aa1b40"
-  integrity sha512-Lzam8lA3XGS90oJ7ubdoHrAusvFETHjywWLkvZe6KJyaMiklj2DyQueZDp6vFGjg4V5aO6CMIEaUWaZjpLj4SQ==
+detox@^17.14.3:
+  version "17.14.9"
+  resolved "https://registry.yarnpkg.com/detox/-/detox-17.14.9.tgz#e269de0937e615eac92ac283b2303751bcf2c072"
+  integrity sha512-WLpl7BJ2xecZJ02hFFWgzo61RnlO/LlsH/U8GWztoXsT0z0t9iKsBci8AhSSErQfH0IHnNtDcZknIxt0gzNhoQ==
   dependencies:
     bunyan "^1.8.12"
     bunyan-debug-stream "^1.1.0"
```

I'll try to pin the actual `17.14.3` version and try it again.

**Comment by fschoenfeldt** [comment]
```diff
diff --git a/package.json b/package.json
index 4b6f351..c6f2842 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
   },
   "devDependencies": {
     "@babel/core": "~7.9.0",
-    "detox": "^17.14.3",
+    "detox": "17.14.3",
     "detox-expo-helpers": "^0.6.0",
     "expo-detox-hook": "^1.0.10",
     "jest": "^26.6.3"
```

Didn't help, too bad.

**Comment by cichun** [comment]
Hi, thanks for your feedback.
I have successfully reproduced this issue🙆‍♂️. 
The dependancies in package.json / yarn.lock seems OK - it works for me.

I think the problem is with default simulator in your system. I am using "iPhone 12 Pro Max" simulator in my everyday work and this simulator is provided in .detoxc.json.
Most people has "iPhone 11" or "iPhone X" simulator as a default. What I mean the default simulator is when you kill Simulator.app on you Mac and run it once again only one device will show up - thats you default simulator.


I have 3 possible solutions for this(lets better call it a hacks):
1) before starting `yarn ios` in first terminal set the default simulator to "iPhone 12 Pro Max"
 so :
 - kill Simulator.app
 - run Simulator.app and then File->Open Simulator->iOS version(eq.14.4)->iPhone 12 Pro Max 
   (source: https://stackoverflow.com/a/63754983/15305253) - now this is you default simulator
 - wait until it runs completly and kill Simulator.app

now do as you did before, so:
 - run `yarn ios` in first terminal ("iPhone 12 Pro Max" should run with sample app)
 - in second terminal run `yarn testGUI`


2) change .detoxc.json to use you system default simulator
 - kill Simulator.app and run it once again. The device that showed up is your default system simulator, I assume it should be "iPhone X"
 - go to .detoxc.json and change "iPhone 12 Pro Max" to "iPhone X" (there are 2 occurencies)

 - now do as you did before, so `yarn ios` and  `yarn testGUI` in second terminal

3) Force Expo to use "iPhone 12 Pro Max"
source: https://stackoverflow.com/questions/47709953/expo-change-default-ios-simulator

- after running `yarn ios` in first terminal type `shift + i` and choose "iPhone 12 Pro Max" from list 
- in second terminal type `yarn testGUI`



Please let me know if any of methods above did work for You.

**Comment by fschoenfeldt** [comment]
First off: Thank you for your extensive answer.

We actually managed to create a working Setup with Detox@18 and Expo@42: https://github.com/fschoenfeldt/clean-expo-detox-testing It's kinda flaky, because we disabled device synchronization and have to wait for the first element to appear but apart from that, tests are working fine.

I'll try to apply your suggestions and post feedback here. At the end of the day, Expo and Detox seem kinda broken and hacky right now.
